content/en/blog/_posts/2025-02-14-cloud-controller-manager-chicken-egg-problem/index.md

```diff
@@ -25,34 +25,27 @@ The [cloud controller manager is part of the control plane][ccm]. It is a critic
 that replaces some functionality that existed previously in the kube-controller-manager and the
 kubelet.
 
-![Components of Kubernetes](https://kubernetes.io/images/docs/components-of-kubernetes.svg)
+{{< figure
+    src="/images/docs/components-of-kubernetes.svg"
+    alt="Components of Kubernetes"
+    caption="Components of Kubernetes"
+>}}
 
 One of the most critical functionalities of the cloud controller manager is the node controller,
 which is responsible for the initialization of the nodes.
 
-As you can see in the following diagram, when the **kubelet** starts, it registers the `Node`
+As you can see in the following diagram, when the **kubelet** starts, it registers the Node
 object with the apiserver, Tainting the node so it can be processed first by the
-cloud-controller-manager. The initial `Node` is missing the cloud-provider specific information,
+cloud-controller-manager. The initial Node is missing the cloud-provider specific information,
 like the Node Addresses and the Labels with the cloud provider specific information like the
 Node, Region and Instance type information.
 
-```mermaid
-sequenceDiagram
-    autonumber
-    rect rgb(191, 223, 255)
-    Kubelet->>+Kube-apiserver: Create Node
-    Note over Kubelet: Taint:<br/> node.cloudprovider.kubernetes.io
-    Kube-apiserver->>-Kubelet: Node  Created
-    end
-    Note over Kube-apiserver: Node is Not Ready<br/> Tainted, Missing Node Addresses*, ...
-    Note over Kube-apiserver: Send Updates
-    rect rgb(200, 150, 255)
-    Kube-apiserver->>+Cloud-controller-manager: Watch: New Node Created
-    Note over Cloud-controller-manager: Initialize Node:<br/>Cloud Provider Labels, Node Addresses, ...
-    Cloud-controller-manager->>-Kube-apiserver: Update Node
-    end
-    Note over Kube-apiserver: Node is Ready
-```
+{{< figure
+    src="ccm-chicken-egg-problem-sequence-diagram.svg"
+    alt="Chicken and egg problem sequence diagram"
+    caption="Chicken and egg problem sequence diagram"
+    class="diagram-large"
```

Comment:
I recommend:
```suggestion
    class="diagram-medium"
```
On my display (2160p) the diagram renders about 1900 pixels tall.